controllers/suggestions/suggest_match_video_controller.py

```diff
@@ -42,11 +43,26 @@ def post(self):
         self._require_login()
 
         match_key = self.request.get("match_key")
-        youtube_url = self.request.get("youtube_url")
-        youtube_id = YouTubeVideoHelper.parse_id_from_url(youtube_url)
-
-        status = SuggestionCreator.createMatchVideoYouTubeSuggestion(self.user_bundle.account.key, youtube_id, match_key)
-
+        video_url = self.request.get("video_url")
+        
+        if any(s in video_url for s in MediaParser.YOUTUBE_URL_PATTERNS):
+            youtube_id = YouTubeVideoHelper.parse_id_from_url(video_url)
+
+            status = SuggestionCreator.createMatchVideoYouTubeSuggestion(self.user_bundle.account.key, youtube_id, match_key)
+        elif any(s in video_url for s in MediaParser.VIMEO_URL_PATTERNS):
+            vimeo_id = None
```

Comment:
This logic is pretty complicated. It feels like we should try to move the logic for searching for suggestions inside of something else so we can have clean, testable functions for each regex searcher.